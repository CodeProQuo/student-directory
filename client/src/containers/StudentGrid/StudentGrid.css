.student-grid {
    max-width: 60%;
    width: 60%;
    max-height: 80%;
    height: 80%;

    border-radius: 10px;
    box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
    background-color: white;

    overflow-y: scroll;
    overflow-x: hidden;
}

.student-grid::-webkit-scrollbar {
    width: 6px;
}

.student-grid::-webkit-scrollbar-track {
    display: none;
}

.student-grid::-webkit-scrollbar-thumb {
    background: #888;
    border-radius: 3px;
    margin: 0 2px 0 0;
    padding: 0 2px 0 0;
}

.student-grid::-webkit-scrollbar-thumb:hover {
    background: #555;
}

p, input {
    margin: 0 0 0 20px;
    font-size: 15px;
    font-family: Raleway, sans-serif;
}

#name-input, #tag-input {
    width: 100%;
    border: 0;
    border-bottom: 1px solid #dcdcdc;
    margin: 5px;
    height: 30px;
    padding: 10px 0 0 5px;
    font-size: 20px;
    font-weight: normal;
}

#name-input:focus, #tag-input:focus {
    outline: none;
    border-bottom: 1px solid black;
}