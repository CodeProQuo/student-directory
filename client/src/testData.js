export const firstStudent = {
    pic: "http://www.google.com",
    firstName: "John",
    lastName: "Doe",
    email: "johndoe@email.com",
    company: "ABC",
    skill: "Archery",
    grades: ["20", "30", "40"],
};

export const secondStudent = {
    pic: "http://www.google.com",
    firstName: "Jane",
    lastName: "Doe",
    email: "janedoe@email.com",
    company: "NBC",
    skill: "Bowling",
    grades: ["50", "60", "70", "80"],
};