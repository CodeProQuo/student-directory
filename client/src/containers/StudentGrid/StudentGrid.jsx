import React, {useState, useEffect} from 'react';
import axios from 'axios';
import Student from "../../components/Student/Student.jsx"
import "./StudentGrid.css";


const StudentGrid = () => {
    const [students, setStudents] = useState([]);
    const [nameFilter, setNameFilter] = useState("");
    const [tagFilter, setTagFilter] = useState("");

    useEffect(() => {
        const queryStudents = () => {
            axios.get(process.env.REACT_APP_URL).then(
                response => {
                    setStudents(response.data);
                });
        };

        queryStudents();
    }, []);

    const studentsJsx = students.map((student, index) => <Student key={index}
                                                                  nameFilter={nameFilter}
                                                                  tagFilter={tagFilter}
                                                                  student={student}/>);

    return (
        <>
            <div className="student-grid">
                <input type="text"
                       id="name-input"
                       onChange={(e) => setNameFilter(e.target.value)}
                       placeholder="Search by name"/>
                <input type="text"
                       id="tag-input"
                       onChange={(e) => setTagFilter(e.target.value)}
                       placeholder="Search by tags"/>
                {studentsJsx}
            </div>
        </>
    );
};

export default StudentGrid;
