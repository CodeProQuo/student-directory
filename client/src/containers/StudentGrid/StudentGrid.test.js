import React from 'react';
import {render, cleanup, fireEvent, waitFor} from '@testing-library/react';
import {URL} from '../../apiCallUrls';
import {rest} from 'msw'
import {setupServer} from 'msw/node';
import StudentGrid from './StudentGrid';
import {firstStudent, secondStudent} from "../../testData";

const server = setupServer(
    rest.get(URL, (req, res, ctx) => {
        return res(ctx.json({students: [firstStudent, secondStudent]}))
    })
);

beforeAll(() => server.listen());
afterEach(() => {
    server.resetHandlers();
    cleanup();
});
afterAll(() => server.close());


it('should take a snapshot of StudentGrid', async () => {
    const {asFragment, getAllByTestId} = render(<StudentGrid/>);
    await waitFor(() => getAllByTestId('student'));

    expect(asFragment(<StudentGrid/>)).toMatchSnapshot();
});


it('should initialize states for StudentGrid as intended', async () => {
    const {getByPlaceholderText, getAllByTestId} = render(<StudentGrid/>);
    await waitFor(() => getAllByTestId('student'));

    expect(getByPlaceholderText('Search by name')).toHaveTextContent("");
    expect(getByPlaceholderText('Search by tags')).toHaveTextContent("");
});

it('should update on input to name filter', async () => {
    const {getByPlaceholderText, getAllByTestId} = render(<StudentGrid/>);

    await waitFor(() => getAllByTestId('student'));

    const nameFilterInput = getByPlaceholderText('Search by name');

    fireEvent.change(nameFilterInput, { target: { value: 'A' } });

    expect(nameFilterInput).toHaveValue('A');
});

it('should update on input to tag filter', async () => {
    const {getByPlaceholderText, getAllByTestId} = render(<StudentGrid/>);

    await waitFor(() => getAllByTestId('student'));
    const tagsFilterInput = getByPlaceholderText('Search by tags');

    fireEvent.change(tagsFilterInput, { target: { value: 'A' } });

    expect(tagsFilterInput).toHaveValue('A');
});
