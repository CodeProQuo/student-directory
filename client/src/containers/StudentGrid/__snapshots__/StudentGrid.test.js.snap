// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of StudentGrid 1`] = `
<DocumentFragment>
  <div
    class="student-grid"
  >
    <input
      id="name-input"
      placeholder="Search by name"
      type="text"
    />
    <input
      id="tag-input"
      placeholder="Search by tags"
      type="text"
    />
    <div
      class="student"
      data-testid="student"
    >
      <div>
        <img
          alt="Student's picture"
          src="http://www.google.com"
        />
      </div>
      <div
        class="student-info"
      >
        <h1>
          JOHN DOE
        </h1>
        <p>
          Email: johndoe@email.com
        </p>
        <p>
          Company: ABC
        </p>
        <p>
          Skill: Archery
        </p>
        <p>
          Average: 30%
        </p>
        <div
          class="student-details collapsed"
        >
          <p>
            Test 1: 20%
          </p>
          <p>
            Test 2: 30%
          </p>
          <p>
            Test 3: 40%
          </p>
          <div
            class="tag-container"
          />
          <input
            id="add-tag-input"
            placeholder="Add new tag"
            type="text"
          />
        </div>
      </div>
      <button
        class="expand-btn closed"
      />
    </div>
    <div
      class="student"
      data-testid="student"
    >
      <div>
        <img
          alt="Student's picture"
          src="http://www.google.com"
        />
      </div>
      <div
        class="student-info"
      >
        <h1>
          JANE DOE
        </h1>
        <p>
          Email: janedoe@email.com
        </p>
        <p>
          Company: NBC
        </p>
        <p>
          Skill: Bowling
        </p>
        <p>
          Average: 65%
        </p>
        <div
          class="student-details collapsed"
        >
          <p>
            Test 1: 50%
          </p>
          <p>
            Test 2: 60%
          </p>
          <p>
            Test 3: 70%
          </p>
          <p>
            Test 4: 80%
          </p>
          <div
            class="tag-container"
          />
          <input
            id="add-tag-input"
            placeholder="Add new tag"
            type="text"
          />
        </div>
      </div>
      <button
        class="expand-btn closed"
      />
    </div>
  </div>
</DocumentFragment>
`;
